solutions/platinum/hr-SubtreesPaths.mdx

```diff
@@ -0,0 +1,196 @@
+---
+id: hr-SubtreesPaths
+source: HR
+title: Subtrees & Paths
+author: Rameez Parwez
+---
+
+Given a rooted tree of $N$ nodes, where each node is uniquely numbered in between $[1...N]$. The node $1$ is the root of the tree. Each node has an integer value which is initially $0$.
+
+You need to perform the following two kinds of queries on the tree:
+- $add$ $t$ $value$: Add $value$ to all nodes in subtree rooted at $t$
+- $max$ $a$ $b$: Report maximum value on the path from $a$ to $b$
```

Comment:
don't summarize the problem statement. explain the solution

Reply:
^^

Reply:
ok

Reply:
unresolving this: can you explain your solution?

Reply:
oops -- didn't realize this was outdated.